internal/usbgadget/utils.go

```diff
@@ -107,6 +110,23 @@ func compareFileContent(oldContent []byte, newContent []byte, looserMatch bool)
 	return false
 }
 
+func writeWithTimeout(file *os.File, data []byte) (n int, err error) {
+	if err := file.SetWriteDeadline(time.Now().Add(hidWriteTimeout)); err != nil {
+		return -1, err
+	}
+
+	n, err = file.Write(data)
+	if err == nil {
+		return
+	}
+
+	if errors.Is(err, os.ErrDeadlineExceeded) {
+		err = nil
+	}
+
+	return
```

Comment:
... so this will return a `nil` for the `err`, effectively, when the deadline expires which means the callers will silently not know that an timeout occurred. That means that there will be nothing logged (might want to at least log something a like 124) AND ALSO we won't trigger the "close the hid file" attempt at recovery (which might be okay, not sure of the intent)